backend/src/main/kotlin/dev/kviklet/kviklet/service/dto/ExecutionRequest.kt

```diff
@@ -277,3 +284,26 @@ data class ExecutionProxy(
 
     override fun getRelated(resource: Resource): SecuredDomainObject? = null
 }
+
+data class SQLDumpResponse(val resource: InputStreamResource, val fileName: String) : SecuredDomainObject {
+    override fun getSecuredObjectId(): String? = fileName
+
+    override fun getDomainObjectType(): Resource = Resource.EXECUTION_REQUEST
+
+    override fun getRelated(resource: Resource): SecuredDomainObject? = null
```

Comment:
I think you just want this to be treated like the execute results, which forward all the implementation to the executionRequest.

The security system was a bit overengineered and I am in the process of simplifying it. So it's a bit confusing atm
For a brief explanation:

- `getSecuredObjectId` this is the id that is checked for permissions. most of the time this should be a connection id. E.g. if the service method is annotated with EXECUTION_REQUEST_GET. The security layer checks if the user has that permission on this specific id. The policies for that are stored in the database and e.g. look like execution_request:get:* (meaning the user is allowed to read requests on all connections).

- ` getDomainObjectType()` should return the type of the object id that is returned, so connection or Execution_Request in this case. But it's not really used anymore right now.

- `getRelated(resource: Resource)` is used to check parent permissions, e.g. previously you'd have permissions on the execution_requests and separate permissions on the connections. Then this was used to return the related resource, so that the system can check if you have all required permissions in the chain on all parent objects (EXECUTE requires GET and CONNECTION_GET for example).

Reply:
Thanks for clarifying it, that makes sense. I've updated the `getSecuredObjectId` in `SQLDumpResponse` to return the connection ID. Let me know if that's good.